src/components/Checkbox/Checkbox.tsx

```diff
@@ -141,13 +165,13 @@ export const Checkbox = React.forwardRef<CheckboxHandles, CheckboxProps>(
             onFocus={onFocus}
             onBlur={onBlur}
             onClick={stopPropagation}
-            onChange={noop}
+            onChange={handleBlur}
```

Comment:
This makes the focus ring not appear when clicking with the mouse. With Radio buttons, this will be tricky because blurring the input prevents the arrows from navigating to the next radio in the group.